src/foam/dao/index/TreePlan.java

```diff
@@ -0,0 +1,25 @@
+package foam.dao.index;
+
+import foam.core.FObject;
+import foam.core.PropertyInfo;
+
+public class TreePlan implements FindPlan {
```

Comment:
TreePlan doesn't seem like a very descriptive name since there will be more than one Tree related Plan. Maybe TreeLookupFindPlan.